### PR TITLE
CB-012: As a Software Engineer, I want to create Login Service so that user can sign up and log in using their user credentials.

### PR DESCRIPTION
**Pull request by sbam1**

### CODE REVIEW

File: src/main/java/org/emela/chautari/domain/UserEntity.java [hunk 1]
```diff
@@ -43,30 +28,96 @@
...
+    }
 
-    private String lastUpdatedUser;
+    public CredentialEntity getCredential() {
+        return credential;
+    }
 
+    public void setCredential(CredentialEntity credential) {
+        this.credential = credential;
+    }
```
Comment (achyutdev) [inline src/main/java/org/emela/chautari/domain/UserEntity.java:122]: Can't we use Lombok annotation for getter and setter (Data)?

File: src/main/java/org/emela/chautari/model/Address.java [hunk 2]
```diff
@@ -1,204 +1,33 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+public class Address {
```
Comment (achyutdev) [inline src/main/java/org/emela/chautari/model/Address.java:13]: are we not replacing with the generated model class from the openAPI service contract?

File: src/main/java/org/emela/chautari/model/AuthenticationResponse.java [hunk 3]
```diff
@@ -1,105 +1,23 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor
+@AllArgsConstructor
+@Data
+@Builder
+public class AuthenticationResponse {
```
Comment (achyutdev) [inline src/main/java/org/emela/chautari/model/AuthenticationResponse.java:13]: generated class?

**Comment by achyutdev** [comment]
Rest LGTM!

File: src/main/java/org/emela/chautari/model/Address.java (hunk 2 above)
Comment (sbam1) [reply src/main/java/org/emela/chautari/model/Address.java:13]: no we are not replacing, right?

File: src/main/java/org/emela/chautari/model/AuthenticationResponse.java (hunk 3 above)
Comment (sbam1) [reply src/main/java/org/emela/chautari/model/AuthenticationResponse.java:13]: I dont have plugin to generate code from open api, so I will merge this code and you can replace with generated ones?

Comment (achyutdev) [reply src/main/java/org/emela/chautari/model/AuthenticationResponse.java:13]: Yes. You can merge now.

**Comment by achyutdev** [comment]
@sbam1 I did not notice earlier that the SetUpApiController has not been implemented from the interface SetUpApi. We have to implement API interface generated from the service contract.  SetupApiController should look like :

```java


@Log4j
@RestController
@RequestMapping("${openapi.chautariRentalService.base-path:/chautari}")
public class SetupApiController implements SetupApi {

    
    private SetupService setupService;

    @Autowired
    public SetupApiController(SetupService setupService) {
        this.setupService = setupService;
    }



    @Override
    public ResponseEntity<SetupResponse> setUpUserAccount(
            @ApiParam(value = "setup request body"  )
            @Valid @RequestBody UserAccountDetail userAccountDetail) {
        log.info("processing new user as: "+ userAccountDetail.getUser().getPerson().getFirstName());
        return new ResponseEntity<>(setupService.setupUser(userAccountDetail), HttpStatus.OK);
    }


}


```

**Pull request merged**